docs/tutorial/hyperfoil.adoc

```diff
@@ -1,129 +1,156 @@
     then:
-    - sh: curl localhost:8080/hello #replace localhost if you used a different ssh computer name
-    - signal: done #tells qDup that the "testing" is done
+      - sh: cd $HYPERFOIL_HOME
+      - add-prompt: "[hyperfoil]$ "
+      - sh: ./bin/cli.sh
+      - add-prompt: "[hyperfoil@in-vm]$ " #do not forget the space
+      - sh: start-local
+      - sh: upload examples/single-request.hf.yaml
+      - sh: run single-request
+      - sh: export 0000 --destination=/tmp/hyperfoil-data/report.json
+      - sh: report 0000 --destination=/tmp/hyperfoil-data/report.html
+      - queue-download: /tmp/hyperfoil-data/report.json
+      - queue-download: /tmp/hyperfoil-data/report.html
+      - signal: done
 
-  ensure-quarkus-cli:
-  - sh: which quarkus
-    then:
-    - regex: no quarkus
-      then: #https://quarkus.io/get-started/
-      - sh: "curl -Ls https://sh.jbang.dev | bash -s - trust add https://repo1.maven.org/maven2/io/quarkus/quarkus-cli/"
-      - sh: "curl -Ls https://sh.jbang.dev | bash -s - app install --fresh --force quarkus@quarkusio"
+  remove-hyperfoil:
+  - sh: cd /tmp && rm -fR hyperfoil*
 
-  getting-started:
+  setup-getting-started:
   - sh: cd /tmp/
   - sh: "[[ ! -d quarkus-quickstarts ]] && git clone https://github.com/quarkusio/quarkus-quickstarts.git"
   - sh: cd quarkus-quickstarts
   - sh: cd getting-started
   - script: ensure-quarkus-cli
+  - sh: export GETTING_STARTED_DIR="$(pwd)" #set an environment variable
+
+  ensure-quarkus-cli:
+  - sh: quarkus version
+    then:
+    - regex: "command not found"
+      then:
+      - sh: "curl -Ls https://sh.jbang.dev | bash -s - trust add https://repo1.maven.org/maven2/io/quarkus/quarkus-cli/"
+      - sh: "curl -Ls https://sh.jbang.dev | bash -s - app install --fresh --force quarkus@quarkusio"
+      - sh: "source ~/.bashrc"
+
+  run-getting-started:
+  - sh: cd $GETTING_STARTED_DIR #ensure we are in the correct directory
   - sh:
       command: quarkus dev
       prompt:
-        "Press [r] to resume testing, [o] Toggle test output, [:] for the terminal, [h] for more options>": "r"
+        "Press [e] to edit command line args (currently ''), [r] to resume testing, [o] Toggle test output, [:] for the terminal, [h] for more options>": "r"
     watch:
-    - regex: "Tests paused"
+    - regex: "Tests completed"
       then:
       - signal: ready
     on-signal:
       done:
-      - ctrlC #exits the process
-
+      - ctrlC
 hosts:
-  test: me@localhost #replace this with your ssh username@hostname
+  test: ${{target-host}}
 roles:
-  example:
+  setup:
     hosts:
     - test
+    setup-scripts:
+    - setup-getting-started
+    run-scripts:
+    - run-getting-started
+  hyperfoil:
+    hosts:
+    - test
+    setup-scripts:
+    - setup-hyperfoil
     run-scripts:
     - test-endpoint
-    - getting-started
+    cleanup-scripts:
+    - remove-hyperfoil
+EOF
+```
+
+Run with and replace `targethost` value with your needs
+```
+targethost=$USER@localhost
```

Comment:
why does the guide introduce the extra step of creating the environment variable? Is this necessary?

Reply:
Changed